nri-redis/src/Redis.hs

```diff
@@ -116,6 +120,12 @@ data Api key a = Api
     --
     -- https://redis.io/commands/ping
     ping :: Internal.Query (),
+    -- | You should probably use `foldWithScan` instead of this function.
```

Comment:
Seeing how complex this is, I wonder if adding `keys` wouldn't be simpler and also solve your use-case.

Reply:
Hmm well I felt the only really complex part was `Redis.Mock`.

And Redis seem to recommend against `keys` in production

> Since these commands allow for incremental iteration, returning only a small number of elements per call, they can be used in production without the downside of commands like [KEYS](https://redis.io/commands/keys) or [SMEMBERS](https://redis.io/commands/smembers) that may block the server for a long time (even several seconds) when called against big collections of keys or elements.

https://redis.io/commands/scan/

Reply:
yeah, I was just considering keys because it's for a mostly dev feature right?

Reply:
Should we only expose `foldWithScan`?

Reply:
Yeah I was debating that too and wondered what you'd think.
Yes I think it's probably better that way.
I'll delete `scan`.